vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -130,4 +138,101 @@ public boolean renameImage(IVSImage image, String newFileName) {
         File renamedFile = new File(path + File.separator + image.getId() + File.separator + newFileName);
         return currentFile.renameTo(renamedFile);
     }
-}
+
+    /**
+     * To create a folder with given name, if it doesn't exist
+     * 
+     * @param folderName - folder name to be created
+     * @return folderName 
+     */
+    @Override
+    public String createFolder(String folderName) {
+        File folder = new File(path + File.separator + folderName);
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+        return folderName;
+    }
+
+    
+    /**
+     * To create a zip of a given folder
+     * 
+     * @param folderName - name of the folder to be zipped
+     * @return byte[] - zipped data as a byte array
+     * @throws IOException if an I/O error occurs while reading files or writing the zip data
+     */
+    @Override
+    public byte[] generateZip(String folderName) throws IOException {
+        Path zipFile = Paths.get(path + File.separator + folderName);
+        ByteArrayOutputStream  byteArrayOutputStream = new ByteArrayOutputStream();
+        Path zipFilePath = Paths.get(File.separator + folderName+".zip");
+        FileOutputStream fileOutputStream = new FileOutputStream(getFile("",folderName+".zip"));
+        try (BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileOutputStream);
+                ZipOutputStream responseZipStream = new ZipOutputStream(bufferedOutputStream);
+                Stream<Path> paths = Files.walk(zipFile)) {
+            paths
+                .filter(path -> !Files.isDirectory(path))
+                .forEach(path -> {
+                    ZipEntry  zipEntry = new ZipEntry(zipFile.relativize(path).toString());
+                    try {
+                        responseZipStream.putNextEntry(zipEntry);
+                        Files.copy(path, responseZipStream);
+                        responseZipStream.closeEntry();
+                        //delete the folder 
+                        Files.delete(zipFile);
+                    } catch (IOException e) {
+                        logger.error("Could not generate Zip folder", e);
```

Comment:
There is something off. This will be done in a loop, so if there is an issue either this will be printed for each folder and then a broken zip file will be returned, or one folder wont' be zipped but the rest will. Likely, we want the whole process to abort if one folder creates an issue but it depends on when the IOException might be thrown.